frontend/src/app/commonComponents/Checkboxes.tsx

```diff
@@ -68,36 +107,42 @@ const Checkboxes = (props: Props) => {
             {errorMessage}
           </div>
         )}
-        <UIDConsumer>
-          {(_, uid) => (
-            <div className="checkboxes">
-              {boxes.map(
-                ({ value, label, disabled, checked, ...inputProps }, i) => (
-                  <div className="usa-checkbox" key={uid(i)}>
-                    <input
-                      className="usa-checkbox__input"
-                      checked={checked}
-                      id={uid(i)}
-                      onChange={onChange}
-                      type="checkbox"
-                      value={value}
-                      name={name}
-                      ref={inputRef}
-                      disabled={disabled || props.disabled}
-                      {...inputProps}
-                    />
-                    <label className="usa-checkbox__label" htmlFor={uid(i)}>
-                      {label}
-                    </label>
-                  </div>
-                )
-              )}
-            </div>
-          )}
-        </UIDConsumer>
+        <div className="grid-row checkboxes">{checkboxesToDisplay}</div>
       </fieldset>
     </div>
   );
 };
 
+type FragmentProps = {
+  boxes: Checkbox[];
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  inputRef?: React.RefObject<HTMLInputElement>;
+  name: string;
+  disabled?: boolean;
+};
+
+const CheckboxesFragment = (props: FragmentProps) => {
+  const { boxes, name, inputRef, onChange } = props;
+
+  return boxes.map(({ value, label, disabled, checked, ...inputProps }) => (
+    <div className="usa-checkbox" key={value}>
```

Comment:
opting to use the `value` here as the key since it'll a stable unique value rather than an on-the-fly generated ID

https://react.dev/learn/rendering-lists#why-does-react-need-keys